app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/WifiDirectManager.java

```diff
@@ -1,482 +0,0 @@
-/*
- * Kiwix Android
- * Copyright (c) 2019 Kiwix <android.kiwix.org>
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program. If not, see <http://www.gnu.org/licenses/>.
- *
- */
-
-package org.kiwix.kiwixmobile.local_file_transfer;
-
-import android.app.Activity;
-import android.content.BroadcastReceiver;
-import android.content.Context;
-import android.content.IntentFilter;
-import android.net.Uri;
-import android.net.wifi.WpsInfo;
-import android.net.wifi.p2p.WifiP2pConfig;
-import android.net.wifi.p2p.WifiP2pDevice;
-import android.net.wifi.p2p.WifiP2pDeviceList;
-import android.net.wifi.p2p.WifiP2pInfo;
-import android.net.wifi.p2p.WifiP2pManager;
-import android.os.AsyncTask;
-import android.os.Build;
-import android.util.Log;
-import android.widget.Toast;
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.net.InetAddress;
-import java.util.ArrayList;
-import javax.inject.Inject;
-import kotlin.Unit;
-import kotlin.jvm.functions.Function0;
-import org.kiwix.kiwixmobile.core.BuildConfig;
-import org.kiwix.kiwixmobile.core.R;
-import org.kiwix.kiwixmobile.core.utils.AlertDialogShower;
-import org.kiwix.kiwixmobile.core.utils.KiwixDialog;
-import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil;
-
-import static android.os.Looper.getMainLooper;
-import static org.kiwix.kiwixmobile.local_file_transfer.FileItem.FileStatus.ERROR;
-import static org.kiwix.kiwixmobile.local_file_transfer.LocalFileTransferActivity.showToast;
-
-/**
- * Manager for the Wifi-P2p API, used in the local file transfer module
- */
-@SuppressWarnings("MissingPermission")
```

Comment:
@yash1ts this is what is missing, I guess the conversion got rid of this suppression